### PR TITLE
Add simple hosts source import/export

### PR DESCRIPTION
**Pull request by RichyHBM**

### CODE REVIEW

File: app/src/main/java/org/adaway/helper/ImportExportHostsHelper.java [hunk 1]
```diff
@@ -0,0 +1,294 @@
...
+            // Get database
+            AppDatabase database = AppDatabase.getInstance(context);
+            // Get input stream from user selected URI
+            try (InputStream inputStream = context.getContentResolver().openInputStream(result)) {
+                if (inputStream != null) {
+                    // Create reader from input stream
+                    try (BufferedReader reader = new BufferedReader(new InputStreamReader(inputStream))) {
+                        Set<String> newHosts = new HashSet<>();
+                        String line = reader.readLine();
+                        while (line != null) {
```
Comment (PerfectSlayer) [inline app/src/main/java/org/adaway/helper/ImportExportHostsHelper.java]: You can have a single line like `while ((line = reader.readLine()) != null) {`

File: app/src/main/java/org/adaway/helper/ImportExportHostsHelper.java [hunk 2]
```diff
@@ -0,0 +1,294 @@
...
+                this.mProgressDialog.dismiss();
+            }
+        }
+
+        private void importHosts(AppDatabase database, Set<String> hosts) {
+            for (String url: hosts) {
+                HostsSource source = new HostsSource();
+                source.setUrl(url);
+                source.setEnabled(true);
+                AppExecutors.getInstance().diskIO().execute(() -> database.hostsSourceDao().insert(source));
```
Comment (PerfectSlayer) [inline app/src/main/java/org/adaway/helper/ImportExportHostsHelper.java]: What happens if the source already exists? I mean, does it crash?

File: app/src/main/java/org/adaway/helper/ImportExportHostsHelper.java [hunk 3]
```diff
@@ -0,0 +1,294 @@
...
+                return false;
+            }
+            // Create export file
+            File exportFile = new File(sdcard, "adaway-hosts-export");
+            // Open writer on the export file
+            try (BufferedWriter writer = new BufferedWriter(new FileWriter(exportFile))) {
+                writer.write(Constants.HEADER_EXPORT + Constants.LINE_SEPARATOR);
+                // Write blacklist items
+                for (HostsSource host : hosts) {
+                    writer.write( host.getUrl() + Constants.LINE_SEPARATOR);
```
Comment (PerfectSlayer) [inline app/src/main/java/org/adaway/helper/ImportExportHostsHelper.java]: Better use `writer.newLine();` than custom `LINE_SEPARATOR` constant (I try to get rid of it).

File: app/src/main/java/org/adaway/helper/ImportExportHostsHelper.java (hunk 1 above)
Comment (RichyHBM) [reply app/src/main/java/org/adaway/helper/ImportExportHostsHelper.java]: I wasnt sure if that would catch the first line? I guess it should do. Will change and push

**Comment by RichyHBM** [comment]
Yes, I am one of the poeple that asked for it :P
I know there are more advanced formats in the Issues suggestions, but I felt like this could be a first step.
Plus I seem to remember seeing that AdAway was getting re-written so maybe this is all that is needed for this version

File: app/src/main/java/org/adaway/helper/ImportExportHostsHelper.java (hunk 2 above)
Comment (RichyHBM) [reply app/src/main/java/org/adaway/helper/ImportExportHostsHelper.java]: No, it handles it.
I tested via creating an export, adding a new host onto the export, and then importing the new file and it was fine. No crash, and no duplicates

**Comment by PerfectSlayer** [comment]
Good remark! Let's ping the related issue here: #919 #950 #990 #999 #1097

Yeah, I'm starting a new version (new UI and a lot of new features). I will merge this PR now and see how users react. If needed, I will improve it in the next version to be a global import / export settings feature.

Thanks again for your contribution! 👏

**Comment by PerfectSlayer** [comment]
I finally improve the feature as a more complete backup solution as it make too many duplicated code.
It will backup hosts sources and user lists in the same file using JSON format.
Next step may be to backup app settings to into the backup file.

**Pull request merged**